### ISSUE TITLE
Shortcodes Handling

### ISSUE THREAD

**Issue by yamilelias** [themosis/themosis#158]
There is a way I can handle the shortcodes? Or do I need to use the native function `addShortcode()` from WordPress?

I searched in the documentation and in google but I didn't find any way to handle shortcuts using the framework. The only thing I found was another issue but in Themosis [Theme repository](https://github.com/themosis/theme), that wasn't helpful at all ([Issue 24](https://github.com/themosis/theme/issues/24)).

If there is any I would recommend that is added to the Documentation site, if not, well is time to develop one, I come from another framework (that is under development right now and lack documentation), but that it handles a function where you can manage your shortcodes inside a controller and it was pretty useful (you can use it as an idea of how to do it).

Thanks by the way.

**Comment by jlambe** [comment]
At the moment, there is no available API to manage shortcodes through a controller. You still have to use the default WordPress function to create it.

Nonetheless, you can currently call a class method in order to build your shortcode using an array syntax:

```php
$class = new CustomClass();
add_shortcode('my-shortcode', [$class, 'methodName']);

// Then from your class file... you can work on your shortcode.
class CustomClass
{
    public function methodName (array $atts)
    {
        // Build your shortcode here...
        // As you have to instantiate your class first, you can also
        // pass any dependencies to the _construct() method as well...
    }
}
```
Otherwise, regarding your request, another user asked the same thing yesterday so why not. We could bring a `CustomClass@methodName` syntax to shortcodes as well saving some typos.

**Comment by yamilelias** [comment]
Yeah it will be cool to have something that could help to save some typos and give a better structure to the code, because (I don't know what do you think) but I've never liked how shortcodes are done haha.

Thank you for your answer, I'll close this topic.

**Comment by yamilelias** [comment]
Yeah it will be cool to have something that could help to save some typos and give a better structure to the code, because (I don't know what do you think) but I've never liked how shortcodes are done haha.

Thank you for your answer, you can close it now.

**Comment by jlambe** [comment]
I'll keep it open for feature record.
